DisguiseUnityRenderStream/Runtime/DisguiseRenderStreamWithClustercs.cs

```diff
@@ -150,81 +156,180 @@ public static ClusterParams ProcessClusterParams(ClusterParams clusterParams)
-            using var clusterKey = Registry.LocalMachine.OpenSubKey("Software\\Unity Technologies\\ClusterDisplay");
-            if (clusterKey != null)
+            try
             {
-                detectedNodeId = true;
-                clusterParams.NodeID = (byte)(int)clusterKey.GetValue("NodeID");
-                clusterParams.RepeaterCount = maybeRepeaterCount ?? (int)clusterKey.GetValue("RepeaterCount");
-                clusterParams.MulticastAddress = (string)clusterKey.GetValue("MulticastAddress");
-                clusterParams.Port = (int)clusterKey.GetValue("MulticastPort");
-                clusterParams.AdapterName = (string)clusterKey.GetValue("AdapterName");
+                clusterParams.NodeID = NegotiateNodeID(clusterParams);
+                ClusterDebug.Log($"Auto-assigning node ID {clusterParams.NodeID} (repeaters: {clusterParams.RepeaterCount})");
+
+                clusterParams.Fence = FrameSyncFence.External;
+                clusterParams.ClusterLogicSpecified = true;
+
+                // Arbitrarily assign Node 0 as the emitter
+                clusterParams.EmitterSpecified = clusterParams.NodeID == 0;
             }
-            else if (maybeRepeaterCount.HasValue)
+            catch (Exception e)
             {
-#endif
-                clusterParams.RepeaterCount = maybeRepeaterCount.Value;
+                Debug.LogException(e);
+                clusterParams.ClusterLogicSpecified = false;
+            }
+
+            return clusterParams;
+        }
 
-                // If we're running several nodes on the same machine, use a single-access file to keep track
-                // of node ids in use.
-                var retries = 0;
-                var nodeIdFilePath = Path.Combine(Path.GetTempPath(), Application.productName + ".node");
-                while (!detectedNodeId && retries < 10)
+        /// <summary>
+        /// Negotiate a distinct ID amongst other running render nodes. Uses the same multi-cast end point
+        /// as the core Cluster Display logic.
+        /// </summary>
+        /// <param name="clusterParams">Parameters containing the networking information.</param>
+        /// <returns></returns>
+        /// <exception cref="TimeoutException">The operation timed out.</exception>
+        /// <exception cref="OperationCanceledException">Multiple errors were encountered. See the log for details.</exception>
+        static byte NegotiateNodeID(ClusterParams clusterParams)
+        {
+            var address = IPAddress.Parse(clusterParams.MulticastAddress);
+            var sendEndPoint = new IPEndPoint(address, clusterParams.Port);
+            var timeoutMilliseconds = (int)clusterParams.HandshakeTimeout.TotalMilliseconds;
+            
+            using var udpClient = new UdpClient();
+            udpClient.EnableMulticast(address,
+                clusterParams.Port,
+                clusterParams.AdapterName,
```

Comment:
Brainstorm: From what I understand we are negotiating on the same clusterParams.AdapterName, which makes perfect sense.  However, if not specified it will "pick one" and use it to negotiate.  This network might then be different (if code diverge in the future) to the one that cluster display will then choose to propagate the state.  Should we put back the adapter name that we found in clusterParams.AdapterName to be sure both are done on the same network?

Reply:
Yes, maybe we can combine this idea with your suggestion for picking the best NIC.